File: app/src/main/java/com/maliotis/petros/stormy/HourlyForecastActivity.java
package com.maliotis.petros.stormy;

import android.content.Intent;
import android.os.Parcelable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.maliotis.petros.stormy.weather.Day;
import com.maliotis.petros.stormy.weather.Hour;

import java.util.Arrays;

public class HourlyForecastActivity extends AppCompatActivity {

    private Hour[] mhours;
    private Day[]  mDays;
    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_hourly_forecast);

        recyclerView = (RecyclerView) findViewById(R.id.recyclerView);

        Intent intent = getIntent();
        Parcelable[] parcelables = intent.getParcelableArrayExtra(MainActivity.HOURLY_FORECAST);
        mhours = Arrays.copyOf(parcelables, parcelables.length, Hour[].class);

        Parcelable[] parcelables1 = intent.getParcelableArrayExtra(MainActivity.HOURLY_DAY_FORECAST);
        mDays = Arrays.copyOf(parcelables1,parcelables1.length,Day[].class);
        HourAdapter hourAdapter = new HourAdapter(this,mhours,mDays);
        recyclerView.setAdapter(hourAdapter);

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);

        recyclerView.setHasFixedSize(true);
    }
}

File: app/src/main/java/com/maliotis/petros/weather/HourlyForecastActivity.kt
package com.maliotis.petros.weather

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.maliotis.petros.weather.weather.Day
import com.maliotis.petros.weather.weather.Hour
import java.util.*

class HourlyForecastActivity : AppCompatActivity() {
    private lateinit var mhours: Array<Hour>
    private lateinit var mDays: Array<Day>
    var recyclerView: RecyclerView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_hourly_forecast)
        recyclerView = findViewById<View>(R.id.recyclerView) as RecyclerView
        val intent = intent
        val parcelables = intent.getParcelableArrayExtra(MainActivity.HOURLY_FORECAST)
        mhours = Arrays.copyOf(parcelables, parcelables.size, Array<Hour>::class.java)
        val parcelables1 = intent.getParcelableArrayExtra(MainActivity.HOURLY_DAY_FORECAST)
        mDays = Arrays.copyOf(parcelables1, parcelables1.size, Array<Day>::class.java)
        val hourAdapter = HourAdapter(this, mhours, mDays)
        recyclerView!!.adapter = hourAdapter
        val layoutManager: RecyclerView.LayoutManager = LinearLayoutManager(this)
        recyclerView!!.layoutManager = layoutManager
        recyclerView!!.setHasFixedSize(true)
    }
}